src/shared/Core/Credential.cs

```diff
@@ -15,21 +18,57 @@ public interface ICredential
         /// Password.
         /// </summary>
         string Password { get; }
+
+        /// <summary>
+        /// The expiry date of the password. This is Git's password_expiry_utc
+        /// attribute. https://git-scm.com/docs/git-credential#Documentation/git-credential.txt-codepasswordexpiryutccode
+        /// </summary>
+        DateTimeOffset? PasswordExpiry { get => null; }
+
+        /// <summary>
+        /// An OAuth refresh token. This is Git's oauth_refresh_token
+        /// attribute. https://git-scm.com/docs/git-credential#Documentation/git-credential.txt-codeoauthrefreshtokencode
+        /// </summary>
+        string OAuthRefreshToken { get => null; }
     }
 
     /// <summary>
     /// Represents a credential (username/password pair) that Git can use to authenticate to a remote repository.
     /// </summary>
-    public class GitCredential : ICredential
+    public record GitCredential : ICredential
```

Comment:
What was the reason for changing this from `class` to `record`?

Reply:
It aided debugging, but not strictly necessary. 

Reverted.